Non-Interactive-Protocol.md

```diff
@@ -0,0 +1,173 @@
+			- MUST replace according to BIP 125
+			- MUST reschedule timer for next bump
+		- MUST NOT spend more in bumping fees than its collateral input value
+
+A node:
+	- if the funding transaction confirms:
+		* MUST track spend of the CET output
+		* MUST advances its off-chain state to the execution phase
+		* MUST note its change output for latter spend
+
+A node:
+	- if the collateral double-spend confirms:
+		* MUST abort the DLC and prevent further interactions with counterparty
+
+## Rationale
+
+Funding transaction must be confirmed to consider the off-chain state initialized. As mempools
+congestion may fluctuate between fees negotiation and transaction propagation, party must adjust
+unilaterally feerate according to their confirmation preferences. A party may choose to wait
+for mempools draining before to adjust the feerate to avoid a fee waste as block height of 
+transition to execution phase doesn't matter, assuming it's before `contract_maturity_bound`.
+
+Reaching near `contract_maturiy_bound` requires to confirm the funding transaction quickly as it's
+now a security matter. After oracle signature release, a malicious party could try to double-spend
+an unconfirmed funding transaction to avoid loosing its collateral.
+
+
+# Execution Phase
+
+The execution phase transitions to the terminal phase either by a CET or a refund
+transaction confirmations.
+
+## Requirements
+
+A node:
+	- if an oracle signature is received:
+		- MUST finalize the CET with a local `SIGHASH_ALL` signature
+		- MUST broadcast the CET
+		- MAY spend its outcome output with a CPFP increasing package feerate to confirm soonly
+		- if the CET is not confirmed and `contract_timeout is 20 blocks ahead of local tip:
+			- MUST rebroadcast the CET
+			- MUST spend its outcome output with a CPFP increasing package feerate to confirm soonly
+			- MUST start a bumping timer of length `security_delay`
+		- if `security_delay` expires:
+			- MUST spend the change output with a replacing CPFP increasing package feerate to confirm soonly
+			- MUST replace according to BIP 125
+			- MUST reschedule timer for next bump
+		- MUST NOT spend more in bumping fees than its outcome value
+
+A node:
+	- if a CET transaction confirms:
+		- MUST note its outcome output for latter spend
+		- MUST advances its off-chain state to terminal phase
+
+A node:
+	- if `contract_timeout` is reached:
+		- MUST finalize the CET with a local `SIGHASH_ALL` signature
+		- MUST broadcast the refund transaction
+		- if the refund is not confirmed after 6 blocks:
+			- MUST spend the change output with a CPFP increasing package feerate to confirm soonly
+			- MUST start a bumping timer of length `congestion_delay`
+		- if `congestion_delay` expires:
+			- MUST spend the change output with a replacing CPFP increasing package feerate to confirm soonly
+			- MUST replace according to BIP 125
+			- MUST reschedule timer for next bump
```

Comment:
What's the issue with the refund not being included in 6 blocks? AFAICT if you reached the refund state you probably have time no?

Reply:
Let's say an oracle is buggy or under attack. If it's key material is compromised after the CET maturation, a malicious CET can be finalized by your counterparty as long as the refund isn't confirmed.

Say differently a lazy oracle key material is toxic until the refund is confirmed.

Reply:
Perhaps it should go from `MUST` to `SHOULD` as it is up to the client how much they trust the oracle? I guess I don't have strong opinions on this though, also it does seem like this is at least a little less high-priority than uses of `congestion_delay` above, perhaps this should be allowed to be a different less strict delay to save on fees?

Reply:
In fact I forget a Rationale for the Execution phase added one.

Note, it's a MUST because ultimately you should manage refund transaction confirmation but `congestion_delay` and feerate are user settings so if you want to spend on fees just pick up values which suits you like `congestion_delay`=100. This was the intention of laying out a Fee bumping policy, let me know if I can make it clearer.